modules/install-dnsmasq/install-dnsmasq

```diff
@@ -98,15 +100,16 @@ function write_consul_config {
   local -r consul_domain="$1"
   local -r consul_ip="$2"
   local -r consul_port="$3"
+  local -r dnsmasq_listen_address="$4"
 
-  log_info "Configuring Dnsmasq to forward lookups of the '$consul_domain' domain to $consul_ip:$consul_port in $CONSUL_DNS_MASQ_CONFIG_FILE"
+  log_info "Configuring Dnsmasq to forward lookups of the '$consul_domain' domain on '$dnsmasq_listen_address' to $consul_ip:$consul_port in $CONSUL_DNS_MASQ_CONFIG_FILE"
   mkdir -p "$DNS_MASQ_CONFIG_DIR"
 
   sudo tee "$CONSUL_DNS_MASQ_CONFIG_FILE" <<EOF
 # Enable forward lookup of the '$consul_domain' domain:
 server=/${consul_domain}/${consul_ip}#${consul_port}
 
-listen-address=${consul_ip}
+listen-address=${dnsmasq_listen_address}
```

Comment:
Hm, it seems like we should keep this at the previous default of `consul_ip`.

That is, in the `install` function, we do the following:

1. By default, `dnsmasq_listen_address` is set to an empty string (`""`).
1. If a user passes in `--dnsmasq-listen-address`, we use their value for `dnsmasq_listen_address`. 
1. Otherwise, after the `while` loop to parse args, if `dnsmasq_listen_address` is still an empty string, then we set it to the `consul_ip` value.

This is fully backwards compatible and still allows `dnsmasq_listen_address` to be customized.

Reply:
Thanks! This is a great way to ensure backward compatibility